src/utils/daemon.rs

```diff
@@ -11,20 +11,33 @@ use crate::{
 use chrono::{Timelike, Utc};
 use failure::Error;
 use log::{debug, error, info};
+use std::process::Command;
 use std::sync::Arc;
 use std::thread;
-use std::time::Duration;
+use std::time::{Duration, Instant};
+
+fn run_git_gc() {
+    let gc = Command::new("git").args(&["gc", "--auto"]).output();
```

Comment:
This needs the path to the repo to clean. I think it'd make most sense to move this to a method on `index::Index` since that's what needs cleaning and has the path available. You can then access it via `docbuilder.index` to call.

Reply:
@Nemo157 inside what method? 

How about `RustwideBuilder::build_package` ? Doesn't it run in the same location as the `git` repo of the crate?

Reply:
A new method, probably just keep this name for it `run_git_gc`.

I have vague ideas of eventually removing the index to queue handling from `docbuilder` completely, it isn't really related to building the docs.

Reply:
@Nemo157 what I meant was, where in the `Index` flow to add it.

 I checked now and the `Index` is called from `RustwideBuilder` which is called from the `Queue`.

So maybe add `last_gc_run` property on the `Queue` and call `run_git_gc` conditionally from inside `build_package`? Something along these lines?

Reply:
I was thinking just keep calling it from here, this is the place that causes us to create new packfiles so should be responsible for knowing when to clean them up.

Reply:
@Nemo157 what is the location you refer to by `calling it from here`?

Reply:
I assume he meant `daemon.rs`. So `fn run_git_gc` gets moved to `Index`, but still gets called from `start_registry_watcher`.

Reply:
@jyn514 @Nemo157 updated, let me know if that what you meant 😊